### PR TITLE
Feature - Story preview endpoint support

### PR DESCRIPTION
**Pull request by marko-ogg**

### CODE REVIEW

File: src/endpoints/Stories/types.ts [hunk 1]
```diff
@@ -180,6 +180,31 @@ export interface SlateStoryUpdateRequest extends GenericUpdateRequest {
...
+export interface PreviewOptions {
+    alignment?: Alignment;
+    appearance?: Appearance;
+}
+
+export enum Alignment {
+    LEFT = 'left',
+    CENTER = 'center',
+    RIGHT = 'right',
+}
```
Comment (e1himself) [inline src/endpoints/Stories/types.ts:201]: This enum is already defined in the `Campaign.ts` file. Let's try to reuse it.

```suggestion
export import Alignment = Campaign.StoryAlignment;
```

File: src/endpoints/Stories/types.ts [hunk 2]
```diff
@@ -180,6 +180,31 @@ export interface SlateStoryUpdateRequest extends GenericUpdateRequest {
...
+
+export enum Appearance {
+    INTRO = 'intro',
+    FULL = 'full',
+}
```
Comment (e1himself) [inline src/endpoints/Stories/types.ts:206]: This enum is already defined in the `Campaign.ts` file. Let's try to reuse it.

```suggestion
export import Appearance = Campaign.StoryAppearance;
```

File: src/endpoints/Stories/types.ts [hunk 3]
```diff
@@ -180,6 +180,31 @@ export interface SlateStoryUpdateRequest extends GenericUpdateRequest {
     content_version?: number;
 }
 
+export interface Preview {
```
Comment (e1himself) [inline src/endpoints/Stories/types.ts:183]: `Response` suffix please.

```suggestion
export interface PreviewResponse {
```

File: src/endpoints/Stories/types.ts (hunk 1 above)
Comment (marko-ogg) [reply src/endpoints/Stories/types.ts]: Oh, good finding. Thanks!

File: src/endpoints/Stories/types.ts [hunk 4]
```diff
@@ -180,6 +181,20 @@ export interface SlateStoryUpdateRequest extends GenericUpdateRequest {
...
+        content: {
+            'text/html': string;
+            'text/plain': string;
+        };
+    };
+}
+
+export interface PreviewOptions {
+    alignment?: Campaign.StoryAlignment;
+    appearance?: Campaign.StoryAppearance;
```
Comment (e1himself) [inline src/endpoints/Stories/types.ts:195]: So no re-exporting? :thinking:

Comment (marko-ogg) [reply src/endpoints/Stories/types.ts]: Oh, ok. Didn't think it's necessary, but if it would be helpful, re-exported. 👍🏻

**Review by e1himself — APPROVED** [review]

File: src/endpoints/Stories/Client.ts [hunk 5]
```diff
@@ -463,4 +465,16 @@ export class Client {
         const url = `${routing.storiesUrl}/${id}`;
         await this.apiClient.delete(url);
     }
+
+    async preview(id: StoryId, options?: PreviewOptions): Promise<PreviewResponse> {
+        const url = `${routing.storiesUrl}/${id}/preview`;
+
+        const { preview } = await this.apiClient.get<{ preview: PreviewResponse }>(url, {
```
[the remark this answers is not in the document]
Comment (marko-ogg) [reply src/endpoints/Stories/Client.ts:472]: Good catch! Thanks!

**Pull request merged**